compiler/rustc_trait_selection/src/solve/alias_relate.rs

```diff
@@ -28,6 +32,12 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
         let tcx = self.tcx();
         let Goal { param_env, predicate: (lhs, rhs, direction) } = goal;
 
+        if self.fast_reject_unnameable_rigid_term(param_env, lhs, rhs)
+            && self.fast_reject_unnameable_rigid_term(param_env, rhs, lhs)
```

Comment:
damn 🤔 if only there was a reason why this fast path didn't fix the hang in rayon...

this literally never triggers 🙃 it needs to be `||`